### PR TITLE
Use `map_overlap` for rolling reductions with Dask

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [ ] Tests added
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] New functions/methods are listed in `api.rst`

This enables Dasks map_overlap for rolling reductions, the [previous issues](https://github.com/pydata/xarray/issues/3165) with memory consumption are now handled in map_overlap as far as I can tell

### CODE REVIEW

**Comment by phofl** [comment]
Looks like CI is green, anything I should check here?

cc @dcherian FYI

File: xarray/core/rolling.py [hunk 1]
```diff
@@ -669,7 +671,10 @@ def _array_reduce(
         if (
             OPTIONS["use_bottleneck"]
             and bottleneck_move_func is not None
-            and not is_duck_dask_array(self.obj.data)
+            and (
+                not is_duck_dask_array(self.obj.data)
+                or module_available("dask", "2024.11.0")
+            )
```
Comment (dcherian) [inline xarray/core/rolling.py:677]: can you delete the comment below please.

**Comment by dcherian** [comment]
I guess this means I can simplify #9720?

File: xarray/core/rolling.py (hunk 1 above)
Comment (phofl) [reply xarray/core/rolling.py:677]: Sorry, thanks!

File: xarray/core/dask_array_ops.py [hunk 2]
```diff
@@ -93,3 +93,15 @@ def _fill_with_last_one(a, b):
         axis=axis,
         dtype=array.dtype,
     )
+
+
+def dask_array_rolling(padded, func, axis, window, min_count=None):
```
[the remark this answers is not in the document]
Comment (phofl) [reply xarray/core/dask_array_ops.py]: I don't think that we need most of it since map_overlap handles that. I adjusted the function

**Comment by phofl** [comment]
The doc build thing is unrelated

**Comment by max-sixty** [comment]
Looks good!

**Comment by phofl** [comment]
Failures seem unrelated

**Pull request merged**